sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/SchemaPruningSuite.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.types._
+
+class SchemaPruningSuite extends SparkFunSuite {
+
+  test("collect struct types") {
+    val datatypes = Seq(
+      IntegerType,
+      ArrayType(IntegerType),
+      StructType.fromDDL("a int, b int"),
+      ArrayType(StructType.fromDDL("a int, b int, c string")),
+      StructType(
+        StructField("a", StructType.fromDDL("a int, b int")) ::
+          StructField("b", IntegerType) :: Nil)
+    )
+
+    val expected = Seq(
+      Seq.empty[StructType],
+      Seq.empty[StructType],
+      Seq(StructType.fromDDL("a int, b int")),
+      Seq(StructType.fromDDL("a int, b int, c string")),
+      Seq(
+        StructType(
+          StructField("a", StructType.fromDDL("a int, b int")) ::
+            StructField("b", IntegerType) :: Nil),
+        StructType.fromDDL("a int, b int"))
+
+    )
+
+    datatypes.zipWithIndex.foreach { case (dt, idx) =>
+      val structs = SchemaPruning.collectStructType(dt, ArrayBuffer.empty[StructType])
+      assert(structs === expected(idx))
+    }
+  }
```

Comment:
Since `pruneDataSchema` becomes public, could you add a test coverage like the following here?
```scala
test("prune data schema") {
  ...
}
```

Reply:
Thank you for adding the test coverage.